### ISSUE TITLE
[BUG] Tabbing out of the game window during gameplay with "Pause game in background" disabled maxes out (100%) a cpu thread until tabbed back into.

### ISSUE THREAD

**Issue by ipaqmaster** [ZDoom/gzdoom#2748]
### GZDoom version

g4.13pre-146-g85ae8f8194-m

### Which game are you running with GZDoom?

Doom 2

### What Operating System are you using?

Linux x86_64

### Please describe your specific OS version

Archlinux. Kernel 6.6.52-1-lts 

### Relevant hardware info

GPU: NVIDIA 2080Ti. CPU: AMD 3900x (12c, 24t)

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

Tabbing out causes the software to reach 100%UTIL on one of the cpu's threads while "Pause game in background" is disabled.

This does not happen if you pause the game or hit Esc for the menu first.

### Steps to reproduce the behaviour.

Explain how to reproduce
1.  Load up doom2.wad (Fresh gzdoom.ini or not)
3. Mash enter to start the game
4. Turn off "Pause game in background" setting.
5. Tab out
6. CPU usage for one of its tasks is now 100%UTIL, stressing out one of the cpu's threads with unnecessary work.

This does not happen when the game gets paused first. It also does not happen  with "Pause game in background"  enabled (Default).

### Your configuration

```ini
Deleted gzdoom.ini for a fresh start (Which used Vulkan + Hardware Accelerated mode).
Bug not present. Realized the game was pausing itself on tab-out and
disabled "Pause game in background" which I found does appear to reproduce the issue.
```

### Provide a Log

NA

**Comment by ipaqmaster** [comment]
This behavior seems to be consistent when downloading `gzdoom-4-12-2-Windows.zip` (GZDoom v4.12.2 | Primary 2024-04-28). Disabling  "Pause game in background" and tabbing out shows gzdoom as a maxed out single thread.

Even after lowering the max frame rate from 500 down to 60 to make sure that wasn't being misrepresented - there is a very clear spike in cpu activity when the game gets tabbed out of during gameplay and "Pause game in background" is disabled.

**Comment by madame-rachelle** [comment]
Can you post your full config? It's not happening for me, I am only spiking at 70% max when I try this with WSL and it doesn't even blip when I run it in native Windows.
![image](https://github.com/user-attachments/assets/2592c207-afff-4094-9770-9f93ef01a983)
![image](https://github.com/user-attachments/assets/04b09f6d-037e-4db8-b772-e8bbfba9a917)

**Comment by madame-rachelle** [comment]
No response, closing

**Comment by RicardoLuis0** [comment]
looking at it, seems like the only difference between windows and the rest with regards to i_pauseinbackground, is that windows changes the process priority to idle, is that possible on linux?

**Comment by RicardoLuis0** [comment]
seems like it's just missing something like:

```cpp
#include <sys/time.h>
#include <sys/resource.h>

[...]

case SDL_WINDOWEVENT_FOCUS_GAINED:
    setpriority(PRIO_PROCESS, 0, 0);
    S_SetSoundPaused(1);
    AppActive = true;
    break;
case SDL_WINDOWEVENT_FOCUS_LOST:
    if(i_pauseinbackground && !noidle && !(sysCallbacks.NetGame && sysCallbacks.NetGame()))
    {
        setpriority(PRIO_PROCESS, 0, 19);
    }
    S_SetSoundPaused(0);
    AppActive = false;
    break;
```

in https://github.com/ZDoom/gzdoom/blob/master/src/common/platform/posix/sdl/sdlglvideo.cpp#L648-L656
<https://linux.die.net/man/2/setpriority>

(won't PR it since i can't really test it tho)

**Comment by RicardoLuis0** [comment]
oop forgot to reopen this, but yeah, this really is a difference between linux and windows

**Comment by ipaqmaster** [comment]
> looking at it, seems like the only difference between windows and the rest with regards to i_pauseinbackground, is that windows changes the process priority to idle, is that possible on linux?

How does that influence the cpu-maxing behavior of the game when the window is not in focus? Wouldn't it still happily do that even if its task proirity gets changed?

**Comment by RicardoLuis0** [comment]
> > looking at it, seems like the only difference between windows and the rest with regards to i_pauseinbackground, is that windows changes the process priority to idle, is that possible on linux?
> 
> How does that influence the cpu-maxing behavior of the game when the window is not in focus? Wouldn't it still happily do that even if its task proirity gets changed?

yeah, can't change that if linux won't stop low-priority processes from hogging the CPU, but it'll still leave CPU time for other processes, unlike now

**Comment by RicardoLuis0** [comment]
can you confirm if this still happens when the new (default true) vid_lowerinbackground cvar is enabled?

**Comment by the-phinet** [comment]
I was unable to test the specific version the user is using, but this does not happen on my system.

Screenshots below:

Current HEAD e0baf7a85
<details>
<summary>Screenshots</summary>

![Image](https://github.com/user-attachments/assets/a004f029-153f-4b51-922d-7698e22fa61b)
</details>

Current HEAD, plus [these](#issuecomment-2409173727) edits
<details>
<summary>Screenshots</summary>

![Image](https://github.com/user-attachments/assets/e0a189a1-f448-49e1-8f47-26056cedc922)
</details>

Current HEAD, plus [these](#issuecomment-2409173727) edits, plus `-noidle` flag
<details>
<summary>Screenshots</summary>

![Image](https://github.com/user-attachments/assets/32d1d5e7-823f-4172-8190-7e04f46aa3cc)
</details>

I assume either it's specific to them or their distro, or the vid_lowerinbackground cvar works :)

**Comment by RicardoLuis0** [comment]
if you can see it happen with vid_lowerinbackground as false then it'd all but confirm it's fixed

**Comment by the-phinet** [comment]
It still happens when pause in background is set to false. So fixed, I think :)

<details>
<summary>Screenshot (vid_lowerinbackground: false, pauseinbackground: true) </summary>

![Image](https://github.com/user-attachments/assets/96187e57-7255-4342-bed2-ec9d96625874)

</details>

<details>
<summary>Screenshot (vid_lowerinbackground: false, pauseinbackground: false)</summary>

![Image](https://github.com/user-attachments/assets/34afd951-19f5-4d44-af20-54c854f3c311)

</details>
